### ISSUE TITLE
Salesforce AI Associate

### ISSUE THREAD

**Issue by su3inni** [su3inni/interested#17]
## 1. AI Fundamentals 
> 17% 

### Machine Learning 
  + learns and adapts based on patterns in data to make accurate predictions 
+ Key characteristic of machine learning
   + uses algorithm to learn from data 
   + and make decisions 
   + **without explicitly programmed** 
   + analyze data identify pattern and make predictions/recommendations 
### Generative AI 
   + Creates new content based on existing data prompted by a human 
### Natural Language Processing : NLP 
   + understand and generate human languages 
   + interaction between computer and human language 
   + speech recognition, machine translation, sentiment analysis, text summarize, chatbot

### Algorithm
   + a simple set of rules for turning an input an output
### Neural Network 
   + a web of connections, guided by weight and biases 

### Key benefit of effective Interaction between humans and ai systems benefits decision making 
   + leads to more informed and balanced decision making 
   + enhance cognitive abilities and performance

**Comment by su3inni** [comment]
### Salesforce Einstein 
https://trailhead.salesforce.com/ko/content/learn/modules/get_smart_einstein_feat

**Comment by su3inni** [comment]
## 3. Ethical considerations of AI 
> 39% 
+ Example of ethical debt : 윤리적 부채
   + launching an AI feature **after discovering a harmful bias**

### Salesforce's Ethical AI Process Maturity Model 
AI 개발/배포 관련된 윤리적이고 책임있는 관행/프로세스 평가 및 개선
+ technical team is assessing the effectiveness of AI development process 
  + use guide the **development of trusted AI solution** 
+ Trusted AI > Process Maturity Model

### ⭐Salesforce Trusted AI Principle > Practices⭐
Guiding ethical & responsible use of AI 
+ Responsible 
+ Accountable 
    본인 행동의 영향을 염두하고 피해나 오류를 피하기 위한 정책 
    seek and leverage feedback for continuous improvement 
    Taking responsibility for one's actions toward customers, partners and society 
   + **Seeking feedback** 
   + mitigate harm 
   + engaging with 3rd party 
   + participate industry
+ Transparent & Explainability 
   + explanation of the prediction's rationale ( user's profile, behavior, preference )
   + a model card : how the model was created 
      + model characteristics & performance ( data source, metrics, limitation ) 
   + understand how AI works , what data is use, what outcome produces 
      + building trust, accountability 
+ Empowering & Education
   + All Skill level to build AI application with clicks, no code 
   + Free AI education with Trailhead 
   + Deliver AI research breakthroughs
+ ⭐ Inclusive & Diversity
  모든 사람의 사회적 가치를 존중하기 ( 다양성 존중 ), 상황에 따라 차별받는 사람이 없도록 보장하는 것이 목표 
   + support **diverse datasets** 
   + consequence scanning Workshops
     제품이나 서비스가 사회에 미치는 영향을 조사하고 피해를 예방하는 과정
   + build inclusive teams
   + Diverse Dataset 
       + models are fair, accurate, representative of diverse customer 
       + prevent & mitigate bias , avoid harmful, discriminatory outcomes 

+ Ethical Challenges 
  + potential for human bias in ml-algorithms 
     + affect ai development process, data collection, data labeling, model selection 
     + train with not presentative diverse customer
  + transparency in decision-making process 

**Bias leads Ethical Challenges** 
+ unfair or discriminatory treatment of customers 
+ violation of customer privacy, erosion of trust 

+ Bias 
  + an inaccuracy that is systematically incorrect in the same direction and affects the success of the person or algorithm that contains it 
+ Disparate Impact 
  + Discriminatory practices toward a particular demographic 
+ Pre-mortem 
  + a pre-project meeting to set measured and realistic expectations and catch the 'what went wrong' before it happens
   + 편향 제거를 위해 
+ 편향 제거를 위해 할 수 있는 것 
   + pre-mortem 진행 
   + 데이터 집합에서 배제되거나 실제보다 비율이 너무 높은 요인 식별 
   + 주기적으로 교육 데이터 평가 
   + Einstein discovery 사용하여 모델의 편향성을 감지하고 방지하기 
### Einstein Discovery 
+ 편향성을 분석할 민감 변수 선택하기 
+ 불평등 효과 해석하기 
+ 프록시 변수 살펴보기 
+ 모델 카드로 예측 정확도 증대하기 

### ⭐Bias Type⭐ 
Bias는 데이터 수집, 분석, 해석에 영향을 미치는 요인 affect quality (reliability of data)
+ Societal Bias : 사회적 편견 
   + data being labeled according to stereotypes or prejutices 
      reinforce existing inequalities, discrimination
+  Confirmation Bias : 확증 편향 
    기존 관념에 따른 확증 편향 
   + from their purchase history 
      + remember informations & existing belief/preference
   + limit diversity , prevent discovering, reduce satisfaction 
+ Gender Bias 

### How to minimize Bias 
+ Implement salesforce's Trusted AI Principles 
   + set of guidelines that help salesforce and customer develop and AI system responsible, ethical and beneficial for all 
+ use **diverse & representative datasets** to train 
+ **test & monitor** for accuracy, reliability, fairness, identify
+ mitigate errors/anomalies/bias 
+ provide **clear & transparent explanations** & sources 
+ use AI to augment & empower human 
+ respect and value the diversity and dignity 
+ do not exclude, discriminate, harm

## 4. Data for AI 
> 36% 

+ Avoid bias from **Demographic** Data 
  + Demograhpic data should be omitted to avoid unintended bias to an AI model 
     + 인구 통계 데이터는 성별, 연령, 인종, 민족, 소득, 교육, 직업과 관련된 개인의 정체성이나 속성에 따라 차별을 일으킬 수 있다. 
  + 각 항목에 따라 데이터 균형을 맞추거나, 마스킹 및 정규화를 진행한다. 
  + 인구 데이터가 아닌 행동 데이터를 사용해야한다. 

### ⭐**Data quality**⭐
   얼마나 현실을 반영하는지, 얼마나 안전하고 일관성있는지, 데이터 오류와 이상값이 얼마나 없는지
  + Key components 
     + Accuracy , completeness, consistency : fundamental 
     + Validity, Uniqueness, Timeliness, Fitness 
  + Key factor influences the ethical use of AI
      + Performance, Accuracy, Reliability
  + **High Quality Data** 
     + Accurate, Complete, Valid, Consistent, Unique, Timely, fit for purpose
     + essential for ensuring unbiased&fair AI decisions 
     + preventing discrimination 
     + trustworthiness of AI-Driven Decision 
     + Benefits of High-Quality data 
        1. improve the reliability and credibility of AI-driven decisions, fostering trust 
        5.  improve customer trust & satisfaction 
           better leads, better understanding of customers, better customer relationships 
        6. enhanced customer value & growth 
             provide more value & opportunities / identify and target their profitable customers ( new & expand good ) 
        7. Increase customer engagement & retention 
              communicate timely & personalized manner 
  + **Poor Quality Data** 
     + produce flawed, misleading results
     + Harm or injustice to someone
     + **lead to biased** : inadvertently learned and amplified 
     + produce erroneous result 
     + reduce accuracy
     + limiting generalization , robustness
     + increasing complexity, opacity 
     + decrease efficiency 

+ Data quality assessment 
  + provides a benchmark for AI predictions : compare predictions/real
  + identify and address data : performance & reliability 
  + data quality assessment 항목 
     + Accuracy 
     + Completeness 
     + Consistency
     + Timeliness

+ Essential **data quality dimension** 
  + Reliability : 정확하고 일관된 정도 
  + Trustworthy, credible

+ **Ensure** Data Quality by field type 
  + Using **Picklist Field Type** 
     + enforcing **consistency**, **accuracy**, **completeness** 
    + by limiting the possible inputs 
    + reducing errors 
    + standard data
  + Using **validation** rule 
     + allows to define **criteria** for data entry 
     + prevent saving records that do not meet the criteria 
+ Page Layout 
   + 일부 레코드에 사용하지 않는 필드 제거 
+ 중복 관리 


+ Ensure Data Quality for AI initiatives
   + collect and curate **high-quality data** from reliable sources 
     + data cleansing
     + data validation 
     + data integration 
     + data monitoring 
   + High-quality data 
      + trustworthy, credible, high-quality from reliable resource 
      + affect performance and accuracy 
      + completeness, correctness, consistency, timeliness, relevance 

+ **Incomplete data quality** ( missing info ) 
   + the accuracy of recommendation is hindered 
   + accuracy , performance, reliability 에 영향을 줌

+ Data quality affects ai models 
  + Data's role in training and fine-tuning salesforce AI models 
  + Einstein Prediction Builder 
  + Einstein Discovery 
  + Einstein Vision & Language 

### Security Principle 
+ Right of Least Privilege > reduce risk of handling sensitive data 
   + by **limiting how many people** have access to data 
      + simplifying management, auditing of data access
   + protect sensitive personal data from unauthorized access, misuse, leakage
+ Data minimization 
   + only necessary & relevant data **should be stored** 
   + decreasing the storage, processing costs, complexity of data 
   + enhancing privacy, protection data by limiting scope 
+ **Data retention** 
   + should be kept **as long as needed**
   + preventing accumulation, duplication of outdated or obsolete data 
   + **improving accuracy and quality data** 
   + respecting rights and preferences of data - who request deletion & modification 

### Data Backup 
key component of **data management strategy** 
+ 백업 빈도, 범위, 방법, 위치 선정 
+ 별도의 위치에 데이터 복사본 생성 및 저장으로 **가용성 보장**
+ Using Cloud service, hard drive, tape 
+ ensure security, availability in case of any loss, corruption, disaster 
+ help recover 

+ Essential performance/relevance of AI model 
  + Volume 
     Volume은 AI 품질 차원이 아니라 모델의 성능이나 관련성에 영향을 주는 요인

+  Protect AI Data Privacy 
  + User Consent 사용자 동의 
     + 어떤 데이터를 수집하는지, 왜 수집하는지, 어떻게 처리하는지, 제 3자 공유 여부 
    + AI infringes on privacy when user consent is not obtained 

+ **Reduce communication inefficiency** with Data Quality 
   + Duplication > confusion, errors, waste 

+ Custom Service Analytics 
   + data should be **fresh and up-to-date** 
   + **age** : **recent data** is important to reflects current situation 

+ How to select **suitable dataset** for **training** an AI model > crucial factor 
  + Size of the dataset 
      affect feasibility 타당성 , quality of model 
  + sufficient information 

+ Data Leakage ( Hindsight bias ) 
  + results from using variables in your prediction model that are known only after the act of interest

+ Data Insights 
  + finding in your data based on thorough analysis generated by Einstein Discovery 

### 데이터 관리 방법 
+ 이름 지정 : 레코드에 대한 명명 규칙 설정 
+ 형식 지정 : 날짜와 비용 표현방식을 지정 
+ 워크플로 : 레코드 생성, 검토, 업데이트 및 보관 프로세스를 결정 
+ 품질 : 레코드를 측정하거나 점수매기는 기능을 포함하여 데이터 품질에 대한 표준을 설정 
  + 기간, 완전성, 사용, 정확성, 일관성 및 중복, 비즈니스에 맞는 다른 품질 
+ 역할 및 소유권 : 레코드를 소유한 사람, 데이터 변경을 책임지는 사람, 데이터가 변경될 때 알림을 수신하는 사람을 결정 
+ 보안 및 권한 : 데이터에 대한 적절한 개인 정보 보호 수준을 결정 
+ 모니터링 : 데이터 품질 관리를 보장하기 위한 프로세스 

## Words 
+ Representaives 담당자 
+ Debt 부채
+ Consent 동의 
+ Maturity 성숙도
+ Demographic 인구 통계 
+ leverage 영향력 
+ Accessment 평가 
+ Reliability , Trustworthy, Credible : 신뢰할 수 있는
+ inadvertently 무심코 
+ amplified 증폭 
+ rationale 이론적 해석

## CHECK 
+ Einstein Prediction Builder 
+ Einstein Discovery ( Sagemaker Canvas 같은 서비스 같다 ) 
  + Requires CRM Anaytics Plus license or Einstien Predictions license ( 추가 비용 필요 ) 
   + 코드가 필요없는 빠른 반복 환경에서 통계적 모델링과 지도형 머신러닝을 통해 비즈니스 인텔리전스를 강화한다. 
   1. 개선할 비즈니스 결과 정하기 ( 회귀분석, 이진분류, 다중클래스 분류 ) 
   2. 분석을 위한 데이터 랭글링 
   3. 데이터를 기반으로 예측모델 생성 
   4. 데이터에 대한 인사이트 제공
   5. 예측 및 개선 사항 생성 
   6. 데이터 알림으로 통찰력 및 모델 개선 
### Salesforce Einstein 
https://trailhead.salesforce.com/ko/content/learn/modules/get_smart_einstein_feat

**Comment by su3inni** [comment]
## 4. Data for AI 
> 36% 

### Avoid bias from **Demographic** Data 
  + Demograhpic data should be omitted to avoid unintended bias to an AI model 
     + 인구 통계 데이터는 성별, 연령, 인종, 민족, 소득, 교육, 직업과 관련된 개인의 정체성이나 속성에 따라 차별을 일으킬 수 있다. 
  + 각 항목에 따라 데이터 균형을 맞추거나, 마스킹 및 정규화를 진행한다. 
  + 인구 데이터가 아닌 행동 데이터를 사용해야한다. 

### ⭐**Data quality**⭐
> 얼마나 현실을 반영하는지, 얼마나 안전하고 일관성있는지, 데이터 오류와 이상값이 얼마나 없는지
+ Key **components** 
  + Accuracy , completeness, consistency : fundamental 
  + Validity, Uniqueness, Timeliness, Fitness 
+ Key factor **influences** the ethical use of AI
  + Performance, Accuracy, Reliability
+ Data quality **assessment** 
  + provides a benchmark for AI predictions : compare predictions/real
  + identify and address data : performance & reliability 
  + data quality assessment 항목 
     + Accuracy 
     + Completeness 
     + Consistency
     + Timeliness

+ Essential **data quality dimension** 
  + Reliability : 정확하고 일관된 정도 
  + Trustworthy, credible

+ How to **Ensure** Data Quality 
   1. By Field type 
      + Using **Picklist Field Type** 
         + enforcing **consistency**, **accuracy**, **completeness** 
        + by limiting the possible inputs 
        + reducing errors 
        + standard data
      + Using **validation** rule 
         + allows to define **criteria** for data entry 
         + prevent saving records that do not meet the criteria 
  2. Page Layout 
     + 일부 레코드에 사용하지 않는 필드 제거 
  3. 중복 관리 
      + **Reduce communication inefficiency** with Data Quality 
         + Duplication > confusion, errors, waste 

### High Quality Data 
+ Accurate, Complete, Valid, Consistent, Unique, Timely, fit for purpose
+ essential for ensuring unbiased&fair AI decisions 
+ preventing discrimination 
+ trustworthiness of AI-Driven Decision 

**Benefits of High-Quality data** 
1. improve the reliability and credibility of AI-driven decisions, fostering trust 
2. improve customer trust & satisfaction 
     better leads, better understanding of customers, better customer relationships 
3. enhanced customer value & growth 
     provide more value & opportunities / identify and target their profitable customers ( new & expand good ) 
5. Increase customer engagement & retention 
     communicate timely & personalized manner 

### Poor Quality Data
+ produce flawed, misleading results
+ Harm or injustice to someone
+ **lead to biased** : inadvertently learned and amplified 
+ produce erroneous result 
+ reduce accuracy
+ limiting generalization , robustness
+ increasing complexity, opacity 
+ decrease efficiency 

### Ensure Data Quality for **AI initiatives**
+ collect and curate **high-quality data** from reliable sources 
  + data cleansing
  + data validation 
  + data integration 
  + data monitoring 
+ **High-quality data** 
  + trustworthy, credible, high-quality from reliable resource 
  + affect performance and accuracy 
  + completeness, correctness, consistency, timeliness, relevance 
+ Data quality affects ai models 
  + Data's role in training and fine-tuning salesforce AI models 
  + Einstein Prediction Builder 
  + Einstein Discovery 
  + Einstein Vision & Language 

**Incomplete data quality** ( missing info ) 
+ the accuracy of recommendation is hindered 
+ accuracy , performance, reliability 에 영향을 줌


### Security Principle 
+ Right of Least Privilege > reduce risk of handling sensitive data 
   + by **limiting how many people** have access to data 
      + simplifying management, auditing of data access
   + protect sensitive personal data from unauthorized access, misuse, leakage
+ Data minimization 
   + only necessary & relevant data **should be stored** 
   + decreasing the storage, processing costs, complexity of data 
   + enhancing privacy, protection data by limiting scope 
+ **Data retention** 
   + should be kept **as long as needed**
   + preventing accumulation, duplication of outdated or obsolete data 
   + **improving accuracy and quality data** 
   + respecting rights and preferences of data - who request deletion & modification 

### Data Backup 
key component of **data management strategy** 
+ 백업 빈도, 범위, 방법, 위치 선정 
+ 별도의 위치에 데이터 복사본 생성 및 저장으로 **가용성 보장**
+ Using Cloud service, hard drive, tape 
+ ensure security, availability in case of any loss, corruption, disaster 
+ help recover 

### Essential performance/relevance of AI model 
  + Volume 
     Volume은 AI 품질 차원이 아니라 모델의 성능이나 관련성에 영향을 주는 요인
+ How to select **suitable dataset** for **training** an AI model > crucial factor 
  + Size of the dataset 
      affect feasibility 타당성 , quality of model 
  + sufficient information 

### Protect AI Data Privacy 
  + User Consent 사용자 동의 
     + 어떤 데이터를 수집하는지, 왜 수집하는지, 어떻게 처리하는지, 제 3자 공유 여부 
    + AI infringes on privacy when user consent is not obtained 

### Custom Service Analytics 
   + data should be **fresh and up-to-date** 
   + **age** : **recent data** is important to reflects current situation 

### Words 
+ Data Leakage ( Hindsight bias ) 
  + results from using variables in your prediction model that are known only after the act of interest

+ Data Insights 
  + finding in your data based on thorough analysis generated by Einstein Discovery 

### 데이터 관리 방법 
+ 이름 지정 : 레코드에 대한 명명 규칙 설정 
+ 형식 지정 : 날짜와 비용 표현방식을 지정 
+ 워크플로 : 레코드 생성, 검토, 업데이트 및 보관 프로세스를 결정 
+ 품질 : 레코드를 측정하거나 점수매기는 기능을 포함하여 데이터 품질에 대한 표준을 설정 
  + 기간, 완전성, 사용, 정확성, 일관성 및 중복, 비즈니스에 맞는 다른 품질 
+ 역할 및 소유권 : 레코드를 소유한 사람, 데이터 변경을 책임지는 사람, 데이터가 변경될 때 알림을 수신하는 사람을 결정 
+ 보안 및 권한 : 데이터에 대한 적절한 개인 정보 보호 수준을 결정 
+ 모니터링 : 데이터 품질 관리를 보장하기 위한 프로세스

**Comment by su3inni** [comment]
+ Representaives 담당자 
+ Debt 부채
+ Consent 동의 
+ Maturity 성숙도
+ Demographic 인구 통계 
+ leverage 영향력 
+ Accessment 평가 
+ Reliability , Trustworthy, Credible : 신뢰할 수 있는
+ inadvertently 무심코 
+ amplified 증폭 
+ rationale 이론적 해석
